testing/acceptance_tests/circuit_test.py

```diff
@@ -0,0 +1,744 @@
+
+        params2_filename = "params2.json"
+        save_circuit_template_params(np.array(params2), params2_filename)
+
+        # When
+        combine_ansatz_params(params1_filename, params2_filename)
+
+        # Then
+        combined_parameters_filename = "combined-params.json"
+        assert os.path.exists(combined_parameters_filename)
+        parameters = load_circuit_template_params(combined_parameters_filename)
+        assert all(parameters == params1 + params2)
+
+        os.remove(params1_filename)
+        os.remove(params2_filename)
+        os.remove(combined_parameters_filename)
+
+
+class TestBuildAnsatzCircuit:
+    @pytest.mark.parametrize(
+        "number_of_layers",
+        [0, 1, 2, 5],
+    )
+    def test_build_ansatz_circuit_with_parameter_values(self, number_of_layers):
+        # Given
+        params = np.random.uniform(low=0, high=np.pi, size=number_of_layers)
+        params_filename = None
+        if params is not None:
+            params_filename = "params.json"
+            save_circuit_template_params(np.array(params), params_filename)
+
+        ansatz_specs = {
+            "module_name": "zquantum.core.interfaces.mock_objects",
+            "function_name": "MockAnsatz",
+            "number_of_layers": number_of_layers,
+            "problem_size": 2,
+        }
+
+        # When
+        build_ansatz_circuit(ansatz_specs=ansatz_specs, params=params_filename)
+
+        # Then
+        circuit_filename = "circuit.json"
+        circuit = load_circuit(circuit_filename)
+        assert isinstance(circuit, Circuit)
+
+        os.remove(circuit_filename)
+        os.remove(params_filename)
+
+    @pytest.mark.parametrize(
+        "number_of_layers",
+        [0, 1, 2, 5],
+    )
+    def test_build_ansatz_circuit_without_parameter_values(self, number_of_layers):
+        # Given
+        ansatz_specs = {
+            "module_name": "zquantum.core.interfaces.mock_objects",
+            "function_name": "MockAnsatz",
+            "number_of_layers": number_of_layers,
+            "problem_size": 2,
+        }
+
+        # When
+        build_ansatz_circuit(ansatz_specs=ansatz_specs)
+
+        # Then
+        circuit_filename = "circuit.json"
+        circuit = load_circuit(circuit_filename)
+        assert isinstance(circuit, Circuit)
+
+        os.remove(circuit_filename)
+
+    def test_build_ansatz_circuit_ansatz_specs_as_string(self):
+        # Given
+        number_of_layers = 2
+        ansatz_specs = {
+            "module_name": "zquantum.core.interfaces.mock_objects",
+            "function_name": "MockAnsatz",
+            "number_of_layers": number_of_layers,
+            "problem_size": 2,
+        }
+
+        # When
+        build_ansatz_circuit(ansatz_specs=json.dumps(ansatz_specs))
+
+        # Then
+        circuit_filename = "circuit.json"
+        circuit = load_circuit(circuit_filename)
+        assert isinstance(circuit, Circuit)
```

Comment:
In ideal world this would be already verified by a static type checker. I wonder if there are other, stronger assertions we might want to verify about the circuit we just loaded?

Reply:
Addressed